<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
  
  <title>NAR Database update 2020 | WJie12</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="DNAproDBDNA–protein complex structure analysisPaperDocument  Update Structure processing upgrades Data update Database and web interface Search Report Integration with Nucleic Acid Database     Discus">
<meta property="og:type" content="article">
<meta property="og:title" content="NAR Database update 2020">
<meta property="og:url" content="https://wjie12.github.io/2020/07/25/NAR2020DatabaseUpdateSummary/index.html">
<meta property="og:site_name" content="WJie12">
<meta property="og:description" content="DNAproDBDNA–protein complex structure analysisPaperDocument  Update Structure processing upgrades Data update Database and web interface Search Report Integration with Nucleic Acid Database     Discus">
<meta property="article:published_time" content="2020-07-25T07:30:10.000Z">
<meta property="article:modified_time" content="2020-07-25T07:30:30.558Z">
<meta property="article:author" content="WJie12">
<meta name="twitter:card" content="summary">
  
  
    <link rel="shortcut icon" href="/favicon.ico">
  
  
<link rel="stylesheet" href="/css/typing.css">

  
<link rel="stylesheet" href="/css/donate.css">

  
  
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/fork-awesome@1/css/fork-awesome.min.css">

  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 4.2.1"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

  
    
      <body>
    
  
      <!-- hexo-inject:begin --><!-- hexo-inject:end --><div id="container" class="container">
        <article id="post-NAR2020DatabaseUpdateSummary" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <header id="header" class="header">
  <nav class="mobile-nav">
    <h1 class="nickname">WJie12</h1>
    <ul class="mobile-nav-menu">
      <label for="mobile-menu-toggle"><a id="menu-button">&#9776; Menu</a></label>
      <input type="checkbox" id="mobile-menu-toggle"/>
      <ul class="mobile-nav-link">
        
        <a href="/">Home</a>
        
        <a href="/archives">Archives</a>
        
        <a href="/about">About</a>
        
      </ul>
    </ul>
  </nav>
	
		<nav id="main-nav" class="main-nav nav-left">
	
	
	  <a class="main-nav-link" href="/">Home</a>
	
	  <a class="main-nav-link" href="/archives">Archives</a>
	
	  <a class="main-nav-link" href="/about">About</a>
	
  </nav>
</header>

  <hr/>
  <div class="article-inner">
    

    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      NAR Database update 2020
    </h1>
  

      </header>
    
    <div class="e-content article-entry typo" itemprop="articleBody">
      
        <h2 id="DNAproDB"><a href="#DNAproDB" class="headerlink" title="DNAproDB"></a><a href="https://dnaprodb.usc.edu">DNAproDB</a></h2><p>DNA–protein complex structure analysis<br><a href="https://academic.oup.com/nar/article/48/D1/D277/5587632">Paper</a><br><a href="https://dnaprodb.usc.edu/documentation.html">Document</a></p>
<ul>
<li>Update<ul>
<li>Structure processing upgrades</li>
<li>Data update</li>
<li>Database and web interface<ul>
<li>Search</li>
<li>Report</li>
<li>Integration with Nucleic Acid Database</li>
</ul>
</li>
</ul>
</li>
<li>Discussion and conclusion<ul>
<li>case study</li>
</ul>
</li>
</ul>
<h2 id="EnhancerAtlas"><a href="#EnhancerAtlas" class="headerlink" title="EnhancerAtlas"></a><a href="http://www.enhanceratlas.org/indexv2.php">EnhancerAtlas</a></h2><p>Enhancers in nine species<br><a href="https://academic.oup.com/nar/article/48/D1/D58/5628925">Paper</a><br><a href="http://www.enhanceratlas.org/helpv2.php">Document</a></p>
<ul>
<li>Materals and methods<ul>
<li>Data sources</li>
<li>Data processing for individual dataset/track</li>
<li>Generation of consensus track</li>
<li>Enhancer-gene interactions</li>
<li>Implenentation of database</li>
</ul>
</li>
<li>Results<ul>
<li>statistics</li>
<li>Database search</li>
<li>Enhancer browser</li>
</ul>
</li>
</ul>
<h2 id="GWAS-Central"><a href="#GWAS-Central" class="headerlink" title="GWAS Central"></a><a href="https://www.gwascentral.org">GWAS Central</a></h2><p>GWAS datasets<br><a href="https://academic.oup.com/nar/article/48/D1/D933/5587637">Paper</a><br><a href="https://help.gwascentral.org/">Document</a></p>
<ul>
<li>Materials and methods<ul>
<li>Data collection and curation</li>
<li>Ontology mapping</li>
<li>Database design</li>
</ul>
</li>
<li>Results<ul>
<li>GWAS data searching</li>
<li>Data visualization</li>
</ul>
</li>
<li>Discussion</li>
</ul>
<h2 id="MatrisomeDB"><a href="#MatrisomeDB" class="headerlink" title="MatrisomeDB"></a><a href="http://matrisome.org">MatrisomeDB</a></h2><p>Extracellular matrix components<br><a href="https://academic.oup.com/nar/article/48/D1/D1136/5581734">Paper</a><br><a href="http://matrisomedb.pepchem.org/tutorial/">Document</a></p>
<ul>
<li>Database infrastructure<ul>
<li>Data sets included</li>
<li>Mass spectrometry data search and processing</li>
</ul>
</li>
<li>Database features and functionalities<ul>
<li>MatrisomeDB query</li>
<li>Hierarchically-clustered protein distribution heatmap</li>
<li>Peptide converage map</li>
</ul>
</li>
<li>Results<ul>
<li>Experimental coverage of the in-silico predicted matrisome</li>
<li>Building an ECM atlas</li>
</ul>
</li>
<li>Future directions<ul>
<li>Future data extension</li>
<li>Extension to include studies from other model organisms</li>
<li>Extracting quantitative data from label-free and label-based proteomics studies</li>
</ul>
</li>
</ul>
<h2 id="MIBiG"><a href="#MIBiG" class="headerlink" title="MIBiG"></a><a href="https://mibig.secondarymetabolites.org">MIBiG</a></h2><p>Biosynthetic Gene Clusters of Known Function<br><a href="https://academic.oup.com/nar/article/48/D1/D454/5587631">Paper</a><br>Document: No</p>
<ul>
<li>Methods and implementation<ul>
<li>Manual curation of entries</li>
<li>Data quality improvements</li>
<li>The new database architecture</li>
</ul>
</li>
<li>Results and discussion<ul>
<li>Data overview<ul>
<li>BGC diversity</li>
<li>Annotation completeness</li>
</ul>
</li>
<li>A new online repository</li>
</ul>
</li>
</ul>
<h2 id="MirGeneDB"><a href="#MirGeneDB" class="headerlink" title="MirGeneDB"></a><a href="www.mirgenedb.org">MirGeneDB</a></h2><p>Animal miRNA complements<br><a href="https://academic.oup.com/nar/article/48/D1/D132/5584683?guestAccessKey=b9fe4339-a36b-49df-96ea-9e2cd7a1c99b">Paper</a><br><a href="https://mirgenedb.org/information">Document</a></p>
<ul>
<li>Expansion of MirGeneDB</li>
<li>Quality of MirGeneDB annotations</li>
<li>Improved web interface of MirGeneDB</li>
<li>MiRNA nomenclature</li>
<li>Functional classfication of MiRNA-seeds</li>
<li>Future developments</li>
</ul>
<h2 id="MSDB"><a href="#MSDB" class="headerlink" title="MSDB"></a><a href="http://data.ccmb.res.in/msdb">MSDB</a></h2><p>Microsatellites from all sequenced genomes<br><a href="https://academic.oup.com/nar/article/48/D1/D155/5584837">Paper</a><br><a href="http://data.ccmb.res.in/msdb/">Document</a></p>
<ul>
<li>Materials and methods<ul>
<li>Data sources</li>
<li>Identification of repeats</li>
<li>Database design</li>
<li>Web interface</li>
</ul>
</li>
<li>Database overview and functionality<ul>
<li>Dashbord view</li>
<li>Species selection</li>
<li>Modal view</li>
<li>Tabel view</li>
<li>Data download</li>
<li>Help page</li>
</ul>
</li>
<li>Discussion<ul>
<li>Case study 1: length preference of AGAT repeats</li>
<li>Case study 2: enrichment of AGC repeats in ruminants</li>
</ul>
</li>
</ul>
<h2 id="Ohnologs"><a href="#Ohnologs" class="headerlink" title="Ohnologs"></a><a href="http://ohnologs.curie.fr">Ohnologs</a></h2><p>Vertebrate ohnologs<br><a href="https://academic.oup.com/nar/article/48/D1/D724/5587630">Paper</a><br><a href="http://ohnologs.curie.fr/cgi-bin/HelpPage.cgi">Document</a></p>
<ul>
<li>Results<ul>
<li>Data collection and processing</li>
<li>Navigating teh OHNOLOGS database</li>
<li>Summary of the contents of the OHNOLOGS database</li>
</ul>
</li>
</ul>
<h2 id="PolyASite"><a href="#PolyASite" class="headerlink" title="PolyASite"></a><a href="http://www.polyasite.unibas.ch">PolyASite</a></h2><p>RNA polyadenylation sites<br><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7145510/">Paper</a><br>Document</p>
<ul>
<li>Materials and methods<ul>
<li>Protocal-spefici pre-processing of reads</li>
<li>Uniform analysis of putative poly(A) sites in individual samples</li>
<li>Clustering of closely spaced sites</li>
<li>Annotation and quantification of pply(A) sites</li>
</ul>
</li>
<li>Results<ul>
<li>Website roadmap</li>
</ul>
</li>
</ul>
<h2 id="WALTZ-DB"><a href="#WALTZ-DB" class="headerlink" title="WALTZ-DB"></a><a href="http://waltzdb.switchlab.org">WALTZ-DB</a></h2><p>Amyloidogenic peptide sequences<br><a href="https://academic.oup.com/nar/article/48/D1/D389/5558128">Paper</a><br><a href="http://waltzdb.switchlab.org/help">Document</a></p>
<ul>
<li>Expanded content and feature improvements<ul>
<li>New peptide entries and database statistics</li>
<li>WALTZ-DB 2.0 novel features</li>
</ul>
</li>
<li>Materials and methods<ul>
<li>Peptide synthesis</li>
<li>Determination of amyloid fibril properties</li>
<li>Structural models and energy calculations</li>
</ul>
</li>
<li>Links to other databases</li>
</ul>

      
      
    </div>
    <footer class="article-footer">
      <ul class="article-meta">
        <li>
          <span class="label">Published Date:</span>
          <a href="/2020/07/25/NAR2020DatabaseUpdateSummary/" class="article-date">
  <time class="dt-published" datetime="2020-07-25T07:30:10.000Z" itemprop="datePublished">2020-07-25</time>
</a>

        </li>
        
          <li>
            <span class="label">Category:</span>
            
  <div class="article-category">
    <a class="article-category-link" href="/categories/other/">other</a>
  </div>


          </li>
        
        
        <hr/>
      </ul>
    </footer>
  </div>
  
    
<nav id="article-nav" class="article-nav">
  
    <span id="article-nav-newer" class="article-nav-link-wrap newer"></span>
  
  
    <a href="/2020/03/09/ConvexOptimization_Introduction/" id="article-nav-older" class="article-nav-link-wrap older">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">ConvexOptimization_Introduction</div>
    </a>
  
</nav>


  
</article>










      </div>
      
    <footer id="footer" class="post-footer footer">
      
        <ul class="footer-links">
          
            <li><a href="/archives/"><span class="fa fa-post"></span></a></li>
          
            <li><a href="https://github.com/WJie12"><span class="fa fa-github-alt"></span></a></li>
          
        </ul>
	    
      <hr/>
      <div id="footerContent" class="footer-content">
        <p>ipsum dolor sit amet, <strong>consectetur adipiscing elit.</strong> Fusce eget urna vitae velit <em>eleifend interdum at ac nisi. In nec ligula lacus. Cum sociis natoque</em> penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed eu cursus erat, ut dapibus quam. Post</p>


      </div>
    </footer>

      








<script src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script>


<script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js"></script>



  
<script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.js"></script>




<script src="/js/typing.js"></script>

<!--[if lt IE 9]>
<script src="https://cdn.jsdelivr.net/npm/html5shiv@3/dist/html5shiv.min.js"></script>
<![endif]-->







    </div><!-- hexo-inject:begin --><!-- hexo-inject:end -->
  </body>
</html>
