<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
  
  <title>Monocle拟时分析 | WJie12</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="读入数据 基于上一篇Seurat的分群的结果进行处理，已经进行过预处理降维分群。 导入上篇保存的pbmc数据 1&gt; load(&#39;.&#x2F;output&#x2F;res0.5.Robj&#39;) 取其中两个亚群进行分析 123&gt; sub_pbmc &lt;- subset(pbmc, idents&#x3D;c(3,4))# 目前数据是个seurat对象&gt; seurat_object &lt;- sub_pb">
<meta property="og:type" content="article">
<meta property="og:title" content="Monocle拟时分析">
<meta property="og:url" content="https://wjie12.github.io/2019/11/13/Monocle/index.html">
<meta property="og:site_name" content="WJie12">
<meta property="og:description" content="读入数据 基于上一篇Seurat的分群的结果进行处理，已经进行过预处理降维分群。 导入上篇保存的pbmc数据 1&gt; load(&#39;.&#x2F;output&#x2F;res0.5.Robj&#39;) 取其中两个亚群进行分析 123&gt; sub_pbmc &lt;- subset(pbmc, idents&#x3D;c(3,4))# 目前数据是个seurat对象&gt; seurat_object &lt;- sub_pb">
<meta property="og:image" content="https://wjie12.github.io/pic/trajectory_state.png">
<meta property="og:image" content="https://wjie12.github.io/pic/trajectory_psudotime.png">
<meta property="og:image" content="https://wjie12.github.io/pic/trajectory_snn_res.0.5.png">
<meta property="og:image" content="https://wjie12.github.io/pic/trajectory_gene.png">
<meta property="og:image" content="https://wjie12.github.io/pic/pseudotime_heatmap.png">
<meta property="og:image" content="https://wjie12.github.io/pic/Rplot_gene_pseudotime.png">
<meta property="og:image" content="https://wjie12.github.io/pic/heatmap_2dir.png">
<meta property="og:image" content="https://wjie12.github.io/pic/branched_pseudotime_2dir.png">
<meta property="article:published_time" content="2019-11-13T09:41:21.000Z">
<meta property="article:modified_time" content="2019-11-13T09:48:39.119Z">
<meta property="article:author" content="WJie12">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wjie12.github.io/pic/trajectory_state.png">
  
  
    <link rel="shortcut icon" href="/favicon.ico">
  
  
<link rel="stylesheet" href="/css/typing.css">

  
<link rel="stylesheet" href="/css/donate.css">

  
  
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/fork-awesome@1/css/fork-awesome.min.css">

  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 4.2.1"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

  
    
      <body>
    
  
      <!-- hexo-inject:begin --><!-- hexo-inject:end --><div id="container" class="container">
        <article id="post-Monocle" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <header id="header" class="header">
  <nav class="mobile-nav">
    <h1 class="nickname">WJie12</h1>
    <ul class="mobile-nav-menu">
      <label for="mobile-menu-toggle"><a id="menu-button">&#9776; Menu</a></label>
      <input type="checkbox" id="mobile-menu-toggle"/>
      <ul class="mobile-nav-link">
        
        <a href="/">Home</a>
        
        <a href="/archives">Archives</a>
        
        <a href="/about">About</a>
        
      </ul>
    </ul>
  </nav>
	
		<nav id="main-nav" class="main-nav nav-left">
	
	
	  <a class="main-nav-link" href="/">Home</a>
	
	  <a class="main-nav-link" href="/archives">Archives</a>
	
	  <a class="main-nav-link" href="/about">About</a>
	
  </nav>
</header>

  <hr/>
  <div class="article-inner">
    

    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      Monocle拟时分析
    </h1>
  

      </header>
    
    <div class="e-content article-entry typo" itemprop="articleBody">
      
        <ol>
<li><p>读入数据</p>
<p>基于上一篇Seurat的分群的结果进行处理，已经进行过预处理降维分群。</p>
<p>导入上篇保存的pbmc数据</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; load(<span class="string">'./output/res0.5.Robj'</span>)</span><br></pre></td></tr></table></figure>
<p>取其中两个亚群进行分析</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt; sub_pbmc &lt;- subset(pbmc, idents=c(<span class="number">3</span>,<span class="number">4</span>))</span><br><span class="line"><span class="comment"># 目前数据是个seurat对象</span></span><br><span class="line">&gt; seurat_object &lt;- sub_pbmc</span><br></pre></td></tr></table></figure>
<p>每个群取10个marker基因，为下面monocle分析准备</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; markers &lt;- FindAllMarkers(sub_pbmc, only.pos = <span class="literal">TRUE</span>, min.pct = <span class="number">0.25</span>, logfc.threshold = <span class="number">0.25</span>)</span><br><span class="line">&gt; top10 &lt;- markers %&gt;% group_by(cluster) %&gt;% top_n(n = <span class="number">10</span>, wt = avg_logFC)</span><br></pre></td></tr></table></figure>
<p>接下来从seurat object中提取创建monocle对象需要的数据</p>
<p>提取表达矩阵，转化为<code>matrix</code>类型</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; data &lt;- as(as.matrix(seurat_object@assays$RNA@data), <span class="string">'sparseMatrix'</span>)</span><br></pre></td></tr></table></figure>
<p>提取细胞表，并转化成<code>AnnotatedDataFrame</code>类型</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; pd &lt;- new(<span class="string">'AnnotatedDataFrame'</span>, data = seurat_object@meta.data)</span><br></pre></td></tr></table></figure>
<p>提取基因注释，并转化成<code>AnnotatedDataFrame</code>类型</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; fData &lt;- data.frame(gene_short_name = row.names(data), row.names = row.names(data))</span><br><span class="line">&gt; fd &lt;- new(<span class="string">'AnnotatedDataFrame'</span>, data = fData)</span><br></pre></td></tr></table></figure>
</li>
<li><p>创建monocle对象</p>
<p><code>newCellDataSet</code>需要输入一个表达矩阵，<code>AnnotatedDataFrame</code>格式的行坐标（sample name）和纵坐标（gene name)</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Construct monocle cds</span></span><br><span class="line">&gt; cds &lt;- newCellDataSet(data,</span><br><span class="line">                         phenoData = pd,</span><br><span class="line">                         featureData = fd,</span><br><span class="line">                         lowerDetectionLimit = <span class="number">0.5</span>,</span><br><span class="line">                         expressionFamily = negbinomial.size());</span><br></pre></td></tr></table></figure>
</li>
<li><p>预估size factors和分散</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; cds &lt;- estimateSizeFactors(cds)</span><br><span class="line">&gt; cds &lt;- estimateDispersions(cds)</span><br></pre></td></tr></table></figure>
</li>
<li><p>提取表达基因</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">expressed_genes &lt;- row.names(subset(fData(cds)))</span><br></pre></td></tr></table></figure>
</li>
<li><p>建立轨迹</p>
<p>采用之前每个群选取的10个marker（top10）来建立轨迹</p>
<p>将top10读取成矩阵</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; ordering_genes&lt;-as.matrix(top10)</span><br><span class="line">ime.txt<span class="string">")</span></span><br></pre></td></tr></table></figure>
<p>过滤</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; cds &lt;- setOrderingFilter(cds, ordering_genes = ordering_genes)</span><br></pre></td></tr></table></figure>
<p>降维</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; cds &lt;- reduceDimension(cds, max_components = <span class="number">2</span>, method = <span class="string">'DDRTree'</span>,auto_param_selection = <span class="literal">F</span>) <span class="comment"># take long time</span></span><br></pre></td></tr></table></figure>
<p>排序</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; cds &lt;- orderCells(cds)</span><br></pre></td></tr></table></figure>
<p>保存</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt; save(cds,file=<span class="string">"./output/orderCells.Robj"</span>)</span><br><span class="line">write.table(pData(cds),file=<span class="string">"./output/my_pseudotime.txt"</span>)</span><br></pre></td></tr></table></figure>
<p>查看</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">&gt; head(pData(cds))</span><br><span class="line">               orig.ident nCount_RNA nFeature_RNA percent.mt RNA_snn_res.0.5 seurat_clusters     S.Score    G2M.Score</span><br><span class="line">AAACATACAACCAC     pbmc3k       <span class="number">2419</span>          <span class="number">779</span>   <span class="number">3.017776</span>               <span class="number">4</span>               <span class="number">4</span>  <span class="number">0.06500339</span> -<span class="number">0.065012414</span></span><br><span class="line">AAACATTGAGCTAC     pbmc3k       <span class="number">4903</span>         <span class="number">1352</span>   <span class="number">3.793596</span>               <span class="number">3</span>               <span class="number">3</span> -<span class="number">0.01906540</span> -<span class="number">0.132349057</span></span><br><span class="line">AAACCGTGTATGCG     pbmc3k        <span class="number">980</span>          <span class="number">521</span>   <span class="number">1.224490</span>               <span class="number">4</span>               <span class="number">4</span> -<span class="number">0.01086219</span>  <span class="number">0.005361576</span></span><br><span class="line">AAACGCTGACCAGT     pbmc3k       <span class="number">2175</span>          <span class="number">782</span>   <span class="number">3.816092</span>               <span class="number">4</span>               <span class="number">4</span> -<span class="number">0.03122878</span> -<span class="number">0.053972321</span></span><br><span class="line">AAACGCTGGTTCTT     pbmc3k       <span class="number">2260</span>          <span class="number">790</span>   <span class="number">3.097345</span>               <span class="number">4</span>               <span class="number">4</span> -<span class="number">0.03190767</span> -<span class="number">0.011423063</span></span><br><span class="line">AAACTTGAAAAACG     pbmc3k       <span class="number">3914</span>         <span class="number">1112</span>   <span class="number">2.631579</span>               <span class="number">3</span>               <span class="number">3</span>  <span class="number">0.04395791</span> -<span class="number">0.080502879</span></span><br><span class="line">               Phase Size_Factor Pseudotime State</span><br><span class="line">AAACATACAACCAC     S   <span class="number">1.3046257</span>   <span class="number">4.343051</span>     <span class="number">6</span></span><br><span class="line">AAACATTGAGCTAC    G1   <span class="number">2.6443075</span>   <span class="number">6.530045</span>     <span class="number">8</span></span><br><span class="line">AAACCGTGTATGCG   G2M   <span class="number">0.5285379</span>   <span class="number">0.000000</span>     <span class="number">1</span></span><br><span class="line">AAACGCTGACCAGT    G1   <span class="number">1.1730306</span>   <span class="number">3.882721</span>     <span class="number">5</span></span><br><span class="line">AAACGCTGGTTCTT    G1   <span class="number">1.2188731</span>   <span class="number">1.263559</span>     <span class="number">1</span></span><br><span class="line">AAACTTGAAAAACG     S   <span class="number">2.1109157</span>   <span class="number">6.543204</span>     <span class="number">8</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>画轨迹图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_cell_trajectory(cds, color_by = <span class="string">"State"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/trajectory_state.png" alt="trajectory_state"></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_cell_trajectory(cds, color_by = <span class="string">"Pseudotime"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/trajectory_psudotime.png" alt="trajectory_psudotime"></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_cell_trajectory(cds, color_by = <span class="string">"RNA_snn_res.0.5"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/trajectory_snn_res.0.5.png" alt="trajectory_snn_res.0.5"></p>
<p>特定基因在在轨迹上的表达量图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_cell_trajectory(cds, markers=<span class="string">"CD79A"</span>, cell_size=<span class="number">0.5</span>, use_color_gradient=<span class="literal">T</span>) + scale_color_gradient2(low=<span class="string">"gray"</span>,mid=<span class="string">"yellow"</span>,high=<span class="string">"red"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/trajectory_gene.png" alt="trajectory_gene"></p>
</li>
<li><p>展示感兴趣的基因热图（一个方向变化）</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&gt; my_pseudotime_de &lt;- differentialGeneTest(cds,fullModelFormulaStr = <span class="string">"~sm.ns(Pseudotime)"</span>,cores = <span class="number">5</span>)</span><br><span class="line"></span><br><span class="line">&gt; sig_gene_names &lt;- row.names(subset(my_pseudotime_de, qval &lt; <span class="number">10</span>^-<span class="number">20</span>))</span><br><span class="line"></span><br><span class="line">&gt; plot_pseudotime_heatmap(cds[sig_gene_names,], num_clusters = <span class="number">4</span>,cores = <span class="number">5</span>,use_gene_short_name = <span class="literal">TRUE</span>,show_rownames = <span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/pseudotime_heatmap.png" alt="pseudotime_heatmap"></p>
</li>
<li><p>展示感兴趣基因基因拟时间表达变化（一个方向</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">&gt; c&lt;-subset(my_pseudotime_de, qval &lt; <span class="number">10</span>^-<span class="number">20</span>)</span><br><span class="line"></span><br><span class="line">&gt; to_be_tested &lt;- row.names(subset(fData(cds),gene_short_name %<span class="keyword">in</span>% c(<span class="string">"CD79A"</span>, <span class="string">"NKG7"</span>, <span class="string">"HLA-DRA"</span>, <span class="string">"CCL5"</span>)))</span><br><span class="line">&gt; cds_subset &lt;- cds[to_be_tested,]</span><br><span class="line"></span><br><span class="line">&gt; diff_test_res &lt;- differentialGeneTest(cds_subset,fullModelFormulaStr = <span class="string">"~sm.ns(Pseudotime)"</span>)</span><br><span class="line"></span><br><span class="line">&gt; diff_test_res[,c(<span class="string">"gene_short_name"</span>, <span class="string">"pval"</span>, <span class="string">"qval"</span>)]</span><br><span class="line"></span><br><span class="line">&gt; plot_genes_in_pseudotime(cds_subset,color_by=<span class="string">"RNA_snn_res.0.5"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/Rplot_gene_pseudotime.png" alt="Rplot_gene_pseudotime"></p>
</li>
<li><p>两个方向的图</p>
<p>热图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_genes_branched_heatmap(cds[sig_gene_names,], num_clusters = <span class="number">4</span>,cores = <span class="number">5</span>,use_gene_short_name = <span class="literal">TRUE</span>,show_rownames = <span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/heatmap_2dir.png" alt="heatmap_2dir"></p>
<p>拟时间表达</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plot_genes_branched_pseudotime(cds_subset, branch_point = <span class="number">2</span>,color_by=<span class="string">"RNA_snn_res.0.5"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/pic/branched_pseudotime_2dir.png" alt="branched_pseudotime_2dir"></p>
</li>
</ol>

      
      
    </div>
    <footer class="article-footer">
      <ul class="article-meta">
        <li>
          <span class="label">Published Date:</span>
          <a href="/2019/11/13/Monocle/" class="article-date">
  <time class="dt-published" datetime="2019-11-13T09:41:21.000Z" itemprop="datePublished">2019-11-13</time>
</a>

        </li>
        
          <li>
            <span class="label">Category:</span>
            
  <div class="article-category">
    <a class="article-category-link" href="/categories/Bioinformatics/">Bioinformatics</a>
  </div>


          </li>
        
        
        <hr/>
      </ul>
    </footer>
  </div>
  
    
<nav id="article-nav" class="article-nav">
  
    <a href="/2019/12/08/smallRNAseq/" id="article-nav-newer" class="article-nav-link-wrap newer">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          Notes for small RNA sequencing analysis
        
      </div>
    </a>
  
  
    <a href="/2019/11/13/Seurat/" id="article-nav-older" class="article-nav-link-wrap older">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Seurat单细胞分群</div>
    </a>
  
</nav>


  
</article>










      </div>
      
    <footer id="footer" class="post-footer footer">
      
        <ul class="footer-links">
          
            <li><a href="/archives/"><span class="fa fa-post"></span></a></li>
          
            <li><a href="https://github.com/WJie12"><span class="fa fa-github-alt"></span></a></li>
          
        </ul>
	    
      <hr/>
      <div id="footerContent" class="footer-content">
        <p>ipsum dolor sit amet, <strong>consectetur adipiscing elit.</strong> Fusce eget urna vitae velit <em>eleifend interdum at ac nisi. In nec ligula lacus. Cum sociis natoque</em> penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed eu cursus erat, ut dapibus quam. Post</p>


      </div>
    </footer>

      








<script src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script>


<script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js"></script>



  
<script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.js"></script>




<script src="/js/typing.js"></script>

<!--[if lt IE 9]>
<script src="https://cdn.jsdelivr.net/npm/html5shiv@3/dist/html5shiv.min.js"></script>
<![endif]-->







    </div><!-- hexo-inject:begin --><!-- hexo-inject:end -->
  </body>
</html>
