<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
  
  <title>Notes for small RNA sequencing analysis | WJie12</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="IntroductionThis pipeline is developed for annotating and profiling small RNAs with UMI attached to both ends. The raw data are reads in FASTQ format. After preprocessing and mapping to reference libr">
<meta property="og:type" content="article">
<meta property="og:title" content="Notes for small RNA sequencing analysis">
<meta property="og:url" content="https://wjie12.github.io/2019/12/08/smallRNAseq/index.html">
<meta property="og:site_name" content="WJie12">
<meta property="og:description" content="IntroductionThis pipeline is developed for annotating and profiling small RNAs with UMI attached to both ends. The raw data are reads in FASTQ format. After preprocessing and mapping to reference libr">
<meta property="og:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/samllrnq.jpg">
<meta property="og:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/s1_y_rnadb_y_genome_result.png">
<meta property="og:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/s3_y_rnadb_y_genome_result.png">
<meta property="og:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/s5_y_rnadb_y_genome_result.png">
<meta property="og:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/n_rnadb_y_genome.png">
<meta property="article:published_time" content="2019-12-08T14:20:42.000Z">
<meta property="article:modified_time" content="2019-12-08T14:33:36.123Z">
<meta property="article:author" content="WJie12">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wjie12.github.io/2019/12/08/smallRNAseq/samllrnq.jpg">
  
  
    <link rel="shortcut icon" href="/favicon.ico">
  
  
<link rel="stylesheet" href="/css/typing.css">

  
<link rel="stylesheet" href="/css/donate.css">

  
  
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/fork-awesome@1/css/fork-awesome.min.css">

  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 4.2.1"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

  
    
      <body>
    
  
      <!-- hexo-inject:begin --><!-- hexo-inject:end --><div id="container" class="container">
        <article id="post-smallRNAseq" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <header id="header" class="header">
  <nav class="mobile-nav">
    <h1 class="nickname">WJie12</h1>
    <ul class="mobile-nav-menu">
      <label for="mobile-menu-toggle"><a id="menu-button">&#9776; Menu</a></label>
      <input type="checkbox" id="mobile-menu-toggle"/>
      <ul class="mobile-nav-link">
        
        <a href="/">Home</a>
        
        <a href="/archives">Archives</a>
        
        <a href="/about">About</a>
        
      </ul>
    </ul>
  </nav>
	
		<nav id="main-nav" class="main-nav nav-left">
	
	
	  <a class="main-nav-link" href="/">Home</a>
	
	  <a class="main-nav-link" href="/archives">Archives</a>
	
	  <a class="main-nav-link" href="/about">About</a>
	
  </nav>
</header>

  <hr/>
  <div class="article-inner">
    

    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      Notes for small RNA sequencing analysis
    </h1>
  

      </header>
    
    <div class="e-content article-entry typo" itemprop="articleBody">
      
        <h1 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h1><p>This pipeline is developed for annotating and profiling small RNAs with UMI attached to both ends. The raw data are reads in FASTQ format. After preprocessing and mapping to reference libraries, we can statistic the reads. Particularly, the reads which can be aligned to the genome but cannot be aligned to RNA sequences databases will be filtered out and written to a CSV format file. The following figure presents a general workflow of the pipeline. </p>
<img src="/2019/12/08/smallRNAseq/samllrnq.jpg" class="" title="workflow">
<h1 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h1><h2 id="Preprocessing"><a href="#Preprocessing" class="headerlink" title="Preprocessing"></a>Preprocessing</h2><p>The preprocessing phase includes three parts: adapter extraction, quality control, UMI extraction. At first, adapters in 3’ end and 5’ end are removed by Cutadapt and then only reads whose length within 27~55nt are retained. Finally, UMIs in both 3’ end and 5’ end are extracted from sequence reads and add to read names with UMI-tools. At the UMI extraction step, reads with poor quality are dropped at the same time by UMI-tools.</p>
<p>Software and parameters used in this phase:</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Software</th>
<th>Parameter</th>
<th>Effect</th>
</tr>
</thead>
<tbody>
<tr>
<td>Cutadapt</td>
<td>-j 40</td>
<td>use 40 CPU cores</td>
</tr>
<tr>
<td></td>
<td>-a AGATCGGAAGAGCACACGTC</td>
<td>remove 3’ end adapter: AGATCGGAAGAGCACACGTC</td>
</tr>
<tr>
<td></td>
<td>-g GTTCAGAGTTCTACAGTCCGACGATC</td>
<td>remove 5’ end adapter: GTTCAGAGTTCTACAGTCCGACGATC</td>
</tr>
<tr>
<td></td>
<td>—overlap=3</td>
<td>require at least three bases match between adapter and read</td>
</tr>
<tr>
<td></td>
<td>—error-rate=0.1</td>
<td>the level of error tolerance (mismatch) in adapter sequences searching</td>
</tr>
<tr>
<td></td>
<td>—times=1</td>
<td>trim no more than one adapter from each reads</td>
</tr>
<tr>
<td></td>
<td>—minimum-length=27 —maximum-length=55</td>
<td>drop reads shorter than 27nt or longer than 55 nt</td>
</tr>
<tr>
<td>UMI-tools extract</td>
<td>—bc-pattern=NNNNNNNNN             —3prime</td>
<td>extract 9nt as UMI from 3’ end</td>
</tr>
<tr>
<td></td>
<td>—bc-pattern=NNNNNN</td>
<td>extract 6nt as UMI from 5’ end</td>
</tr>
<tr>
<td></td>
<td>—quality-filter-threshold 20</td>
<td>filter low quality reads</td>
</tr>
</tbody>
</table>
</div>
<h2 id="Mapping"><a href="#Mapping" class="headerlink" title="Mapping"></a>Mapping</h2><p>The preprocessed reads are still in FASTQ format. The reads are aligned to genome and RNA databases with Bowtie, respectively. The following table lists the reference libraries we use in alignment. After aligning to references, the duplicate reads (reads with the same alignment position and UMI) will be dropped by umi_tools. After mapping, we can get the annotated reads in SAM format.</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Genome</th>
<th>RNA databases(sort by mapping order)</th>
</tr>
</thead>
<tbody>
<tr>
<td>hg19</td>
<td>mature tRNA</td>
</tr>
<tr>
<td></td>
<td>tRNA</td>
</tr>
<tr>
<td></td>
<td>rRNA</td>
</tr>
<tr>
<td></td>
<td>miRNA</td>
</tr>
<tr>
<td></td>
<td>piRNA</td>
</tr>
<tr>
<td></td>
<td>Rfam</td>
</tr>
</tbody>
</table>
</div>
<p>Software and parameters used in this phase:</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Software</th>
<th>Parameter</th>
<th>Effect</th>
</tr>
</thead>
<tbody>
<tr>
<td>Bowtie</td>
<td>-q</td>
<td>input format: FASTQ</td>
</tr>
<tr>
<td></td>
<td>—threads 40</td>
<td>run in 40 threads</td>
</tr>
<tr>
<td></td>
<td>-v 1</td>
<td>allow 1 mismatch</td>
</tr>
<tr>
<td></td>
<td>-m 1</td>
<td>each reads can only be aligned once</td>
</tr>
<tr>
<td></td>
<td>-a</td>
<td>keep all mapping results</td>
</tr>
<tr>
<td></td>
<td>-un</td>
<td>write all reads that could not be aligned to a file</td>
</tr>
<tr>
<td>UMI-tools dedup</td>
<td>—method=unique</td>
<td>do not consider the sequencing mistakes in UMIs</td>
</tr>
<tr>
<td></td>
<td>—read-length</td>
<td>duplicate reads must have the same length</td>
</tr>
</tbody>
</table>
</div>
<h2 id="Analysis"><a href="#Analysis" class="headerlink" title="Analysis"></a>Analysis</h2><p>With the SAM format files in the mapping phase, we can count the reads. The reads with the same annotation will be counted as the same RNA. Pie graphs to visualize the percentage of different types of RNAs are plotted based on the counts. Besides counting the reads that mapping to the RNA databases, we can also filter the sequences that can be aligned to the genome but not to RNA databases. The statistics results are saved in y_rnadb_y_genome.csv and n_rnadb_y_genome.csv. The statistic and visualization are conducted with Python scripts. </p>
<h1 id="Implement"><a href="#Implement" class="headerlink" title="Implement"></a>Implement</h1><p>The preprocessing phase and mapping phase are implemented by Shell scripts, and the analysis phase is implemented by Python scripts. The source code and more details of these scripts can be found in <a href="https://github.com/WJie12/SmallRNASeq">SmallRNASeq</a>. </p>
<p>Full analysis results of samples can be found in the attachment.</p>
<h1 id="Results"><a href="#Results" class="headerlink" title="Results"></a>Results</h1><p>The following pie graphs show the mapping results of different samples.</p>
<p>Sample 1:</p>
<img src="/2019/12/08/smallRNAseq/s1_y_rnadb_y_genome_result.png" class="" title="s1_y_rnadb_y_genome_result">
<p>Sample 3:</p>
<img src="/2019/12/08/smallRNAseq/s3_y_rnadb_y_genome_result.png" class="" title="s3_y_rnadb_y_genome_result">
<p>Sample 5:</p>
<img src="/2019/12/08/smallRNAseq/s5_y_rnadb_y_genome_result.png" class="" title="s5_y_rnadb_y_genome_result">
<p>The head of the n_rnadb_y_genome.csv for sample 1:</p>
<img src="/2019/12/08/smallRNAseq/n_rnadb_y_genome.png" class="" title="n_rnadb_y_genome">

      
      
    </div>
    <footer class="article-footer">
      <ul class="article-meta">
        <li>
          <span class="label">Published Date:</span>
          <a href="/2019/12/08/smallRNAseq/" class="article-date">
  <time class="dt-published" datetime="2019-12-08T14:20:42.000Z" itemprop="datePublished">2019-12-08</time>
</a>

        </li>
        
          <li>
            <span class="label">Category:</span>
            
  <div class="article-category">
    <a class="article-category-link" href="/categories/Bioinformatics/">Bioinformatics</a>
  </div>


          </li>
        
        
        <hr/>
      </ul>
    </footer>
  </div>
  
    
<nav id="article-nav" class="article-nav">
  
    <a href="/2020/01/19/DGM4SCA/" id="article-nav-newer" class="article-nav-link-wrap newer">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          Deep Generative Model for Auto-annotation in Single-cell Analysis
        
      </div>
    </a>
  
  
    <a href="/2019/11/13/Monocle/" id="article-nav-older" class="article-nav-link-wrap older">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Monocle拟时分析</div>
    </a>
  
</nav>


  
</article>










      </div>
      
    <footer id="footer" class="post-footer footer">
      
        <ul class="footer-links">
          
            <li><a href="/archives/"><span class="fa fa-post"></span></a></li>
          
            <li><a href="https://github.com/WJie12"><span class="fa fa-github-alt"></span></a></li>
          
        </ul>
	    
      <hr/>
      <div id="footerContent" class="footer-content">
        <p>ipsum dolor sit amet, <strong>consectetur adipiscing elit.</strong> Fusce eget urna vitae velit <em>eleifend interdum at ac nisi. In nec ligula lacus. Cum sociis natoque</em> penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed eu cursus erat, ut dapibus quam. Post</p>


      </div>
    </footer>

      








<script src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script>


<script src="https://cdn.jsdelivr.net/npm/clipboard@2/dist/clipboard.min.js"></script>



  
<script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3/dist/jquery.fancybox.min.js"></script>




<script src="/js/typing.js"></script>

<!--[if lt IE 9]>
<script src="https://cdn.jsdelivr.net/npm/html5shiv@3/dist/html5shiv.min.js"></script>
<![endif]-->







    </div><!-- hexo-inject:begin --><!-- hexo-inject:end -->
  </body>
</html>
